### PR TITLE
Add links for learning more (part 2 of 4)

### LINKED ISSUE #161: Add links for learning more (Part 2 of 4)
## Background

When the Super Rentals tutorial was revamped for Octane edition, many pages in the [Ember Guides](https://guides.emberjs.com/) were in flux. Possibly as a result, the tutorial (this repo) contains many link placeholders, marked with the keyword `TODO`. The links are meant to provide additional learning resources to the reader.

## Problem

Let's fill in links for the following pages in the tutorial:

- `04-component-basics`
- `05-more-about-components`
- `06-interactive-components`

You can find the Markdown files in the `src/markdown/tutorial` directory. For each TODO listed below, please replace the pattern `*[description][TODO: add a link]*` with `*[description](link)*`. The latter is how we add a link in Markdown.

When the link has the domain of `guides.emberjs.com`, please add a relative path instead:

```
Instead of:
[route](https://guides.emberjs.com/release/routing/defining-your-routes/)

Write (must include trailing slash):
[route](../../../routing/defining-your-routes/)
```

### 04-component-basics

- [ ] Line 7. `link to refactor` → https://guides.emberjs.com/release/components/introducing-components/#toc_breaking-it-into-pieces
- [ ] Line 23. `link to component` → https://guides.emberjs.com/release/components/introducing-components/
- [ ] Line 57. `link to content` → https://guides.emberjs.com/release/components/block-content/
- [ ] Line 75. `link to refactored` → https://guides.emberjs.com/release/components/introducing-components/#toc_breaking-components-down-further
- [ ] Line 57. `link to content` → https://guides.emberjs.com/release/components/block-content/
- [ ] Line 142. `link to encapsulating` → https://guides.emberjs.com/release/components/component-arguments-and-html-attributes/
- [ ] Line 156. `link to component test` → https://guides.emberjs.com/release/testing/testing-components/
- [ ] Line 156. Replace `*[rendering
 test][TODO: link to rendering test]*` with a normal text `rendering test`
- [ ] Line 356. `add anchor link back to where we first mentioned it` → `#toc_passing-content-to-components-with-yield`

### 05-more-about-components

- [ ] Line 165. `link to splattributes` → https://guides.emberjs.com/release/components/component-arguments-and-html-attributes/#toc_html-attributes

### 06-interactive-components

- [ ] Line 29. `link to behavior` → https://developer.mozilla.org/docs/Learn/JavaScript/Building_blocks/Events
- [ ] Line 33. `link to states` → https://guides.emberjs.com/release/components/component-state-and-actions/
- [ ] Line 46. Fix typo: `app/components/rentals/image.js` → `app/components/rental/image.js`
- [ ] Line 46. `link to JavaScript class` → https://javascript.info/class
- [ ] Line 46. `link to inheriting` → https://javascript.info/class-inheritance
- [ ] Line 50. `link to Glimmer component` → https://guides.emberjs.com/release/upgrading/current-edition/glimmer-components
- [ ] Line 52. `link to classic components` → https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/
- [ ] Line 90. `link to conditionals` → https://guides.emberjs.com/release/components/conditional-content/
- [ ] Line 143. `link to method` → https://guides.emberjs.com/release/in-depth-topics/native-classes-in-depth/#toc_methods
- [ ] Line 163. `link to decorator` → https://guides.emberjs.com/release/in-depth-topics/native-classes-in-depth/#toc_decorators
- [ ] Line 165. `link to blocks` → https://guides.emberjs.com/release/components/conditional-content/#toc_block-if
- [ ] Line 206. `link to modifier` → https://guides.emberjs.com/release/components/template-lifecycle-dom-and-modifiers/#toc_event-handlers
- [ ] Line 284. `link to expression` → https://guides.emberjs.com/release/components/conditional-content/#toc_inline-if
- [ ] Line 307. Replace `*[condition][TODO: link to condition]*` with normal text `condition`

### PR DESCRIPTION
**Pull request by AndriyTykhan**
This pull request closes #161.

### CODE REVIEW

**Review by ijlee2 — CHANGES_REQUESTED** [review]
Hi, @AndriyTykhan. Thank you for taking an interest in helping out with the Super Rentals tutorial!

I noticed in `04-component-basics` that the links are formatted incorrectly. As a result, I think the CI's build checks are failing.

Will you review the 3 files that you've edited and check that the links are formatted according to the description in the [issue](https://github.com/ember-learn/super-rentals-tutorial/issues/161#issue-718487975)?

To see examples of how to replace a placeholder link, I recommend looking at https://github.com/ember-learn/super-rentals-tutorial/pull/165/files

If you have questions, please feel free to comment in this pull request or in the issue. Thanks again!

File: src/markdown/tutorial/part-1/04-component-basics.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 ember server
 ```
 
-In this chapter, you will *[refactor][TODO: link to refactor]* your existing templates to use components. We will also be adding a site-wide navigation bar:
+In this chapter, you will *[refactor](https://guides.emberjs.com/release/components/introducing-components/#toc_breaking-it-into-pieces)* your existing templates to use components. We will also be adding a site-wide navigation bar:
```
Comment (ijlee2) [inline src/markdown/tutorial/part-1/04-component-basics.md:7]:
```suggestion
In this chapter, you will [refactor](../../../components/introducing-components/#toc_breaking-it-into-pieces) your existing templates to use components. We will also be adding a site-wide navigation bar:
```

**Comment by ijlee2** [comment]
@AndriyTykhan I made a mistake with the issue description. When adding the link, will you leave the `*` as is?

`*[description][TODO: add a link]*` becomes `*[description](link)*`

File: src/markdown/tutorial/part-1/04-component-basics.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ In doing so, you will learn about:
 
 ## Extracting Markup into Components
 
-In a [previous chapter](../building-pages/), we got a light introduction to *[components][TODO: link to components]* when using `<LinkTo>` to connect our pages. To recap, we said that components are Ember's way of creating *[custom tags][TODO: link to custom tags]* to supplement the built-in HTML tags from the browser. Now, we are going to create our own components!
+In a [previous chapter](../building-pages/), we got a light introduction to *[components][(https://guides.emberjs.com/release/components/introducing-components/)* when using `<LinkTo>` to connect our pages. To recap, we said that components are Ember's way of creating *[custom tags][TODO: link to custom tags]* to supplement the built-in HTML tags from the browser. Now, we are going to create our own components!
```
Comment (ijlee2) [inline src/markdown/tutorial/part-1/04-component-basics.md:23]:
```suggestion
In a [previous chapter](../building-pages/), we got a light introduction to *[components](../../../components/introducing-components/)* when using `<LinkTo>` to connect our pages. To recap, we said that components are Ember's way of creating *[custom tags][TODO: link to custom tags]* to supplement the built-in HTML tags from the browser. Now, we are going to create our own components!
```

**Comment by AndriyTykhan** [comment]
@ijlee2 Hi, I've made changes that you recommend and check files, it looks good for me, but build fails and I am not sure why, could you help with it?

**Comment by ijlee2** [comment]
@AndriyTykhan Yep, I will have a look at the files. Thanks for working on the issue again!

File: src/markdown/tutorial/part-1/04-component-basics.md [hunk 3]
```diff
@@ -54,7 +54,7 @@ That's it, we have created our first component! We can now *[invoke][TODO: link
 
 ## Passing Content to Components with `{{yield}}`
 
-When invoking a component, Ember will replace the component tag with the content found in the component's template. Just like regular HTML tags, it is common to pass *[content][TODO: link to content]* to components, like `<Jumbo>some content</Jumbo>`. We can enable this using the `{{yield}}` keyword, which will be replaced with the content that was passed to the component.
+When invoking a component, Ember will replace the component tag with the content found in the component's template. Just like regular HTML tags, it is common to pass *[content](https://guides.emberjs.com/release/components/block-content/)* to components, like `<Jumbo>some content</Jumbo>`. We can enable this using the `{{yield}}` keyword, which will be replaced with the content that was passed to the component.
```
Comment (ijlee2) [inline src/markdown/tutorial/part-1/04-component-basics.md:57]:
```suggestion
When invoking a component, Ember will replace the component tag with the content found in the component's template. Just like regular HTML tags, it is common to pass *[content](../../../components/block-content/)* to components, like `<Jumbo>some content</Jumbo>`. We can enable this using the `{{yield}}` keyword, which will be replaced with the content that was passed to the component.
```

**Review by ijlee2 — APPROVED** [review]
Hi, @AndriyTykhan. I worked on your branch directly so that I could find and fix Markdown lint errors. It turned out there were a couple of extra characters that caused linting to fail:

- [04-component-basics, line 23 had an extra `[` character](https://github.com/ember-learn/super-rentals-tutorial/pull/166/commits/1724f67b0565ff30fadf254cd35fcb6642a36649#diff-cfd7e794f31fbc0e09db790f6d0d7df296363de1721179cbf2807179b44d576bL23)
- [06-interactive-components, line 284 had an extra `\` character](https://github.com/ember-learn/super-rentals-tutorial/pull/166/commits/11dd90dfc61c01ca755611427d7c9baf1956857b#diff-96fd485ed873422b1a6c86cc7381dcf3a7b28f788f57e8a9d5bc05ef05009b2fL284)

I also updated your branch so that all links to `guides.emberjs.com` have a relative path. Your pull request can now be merged.

Thanks again for working with me in figuring out the problem. Congratulations on making your 1st contribution to the Super Rentals tutorial! 🎉

**Pull request merged**